### PR TITLE
Add UI feedback

### PR DESCRIPTION
**Pull request by thecalcc**
SDCP-854

### CODE REVIEW

File: scripts/apps/highlights/services/HighlightsService.ts [hunk 1]
```diff
@@ -111,7 +114,25 @@ export function HighlightsService(api, $q, $cacheFactory, packages: IPackagesSer
      * Mark an item for a highlight
      */
     service.markItem = function(highlight, markedItem) {
-        return api.save('marked_for_highlights', {highlights: [highlight], marked_item: markedItem._id});
+        dispatchEvent(new CustomEvent('highlights-loading', {
```
Comment (tomaskikutis) [inline scripts/apps/highlights/services/HighlightsService.ts]: Can you abstract it into a function so if you want to use it for some other action you would only need to call a function, but wouldn't need to know about events?

```
function trackArticleActionProgress<T>(p: () => Promise<T>, successMessage: string, errorMessage: string): Promise<T> {
    //
}

function markItem(highlight, markedItem) {
    return trackArticleActionProgress(
        () => api.save(
            'marked_for_highlights',
            {highlights: [highlight], marked_item: markedItem._id},
        ),
        gettext('Item marked'),
        gettext('Couldn\'t mark item'),
    );
}

```

File: scripts/apps/search/components/Item.tsx [hunk 2]
```diff
@@ -119,13 +120,21 @@ export class Item extends React.Component<IProps, IState> {
...
     }
 
+    handleHighlightsLoading(e: CustomEvent) {
+        if (e.detail.itemId === this.props.item._id) {
+            this.setState({loading: e.detail.loading});
+        }
+    }
+
     componentDidMount() {
+        addEventListener('highlights-loading', this.handleHighlightsLoading);
```
Comment (tomaskikutis) [inline scripts/apps/search/components/Item.tsx]: Let's not couple it to highlights here. Just call it loading indicator. That way we'll be able to reuse it for other actions and will not be lying via naming.

File: scripts/core/helpers/network.tsx [hunk 3]
```diff
@@ -173,3 +174,30 @@ export function uploadFileWithProgress<T>(
             });
         });
 }
+
+export function trackArticleActionProgress<T>(
+    p: () => Promise<T>,
```
Comment (tomaskikutis) [inline scripts/core/helpers/network.tsx]: `getPromise` would be a better name. `p` was a placeholder in my initial snippet.

File: scripts/core/helpers/network.tsx [hunk 4]
```diff
@@ -173,3 +174,30 @@ export function uploadFileWithProgress<T>(
...
+    itemId: string,
+    successMessage: string,
+    errorMessage: string,
+): Promise<T> {
+    dispatchEvent(new CustomEvent('action-loading', {
```
Comment (tomaskikutis) [inline scripts/core/helpers/network.tsx]: would be good to call it `article-action-loading` so if anyone is logging events they would discover quicker from which part is it coming from.

**Review by tomaskikutis — APPROVED** [review]

**Comment by thecalcc** [comment]
cherry pick to release/2.8 and update cp dev

**Pull request merged**